### ISSUE TITLE
DOC Replace a sklearn classifier in an example with a better suited one

### ISSUE THREAD

**Issue by TamaraAtanasoska** [fairlearn/fairlearn#1498]
#### Describe the issue linked to the documentation
`sklearn.ensemble.HistGradientBoostingClassifier` is a better, more modern choice than `sklearn.ensemble.RandomForestClassifier` for https://fairlearn.org/v0.12/user_guide/mitigation/reductions.html#exponentiated-gradient. 

[Related discussion source](https://github.com/fairlearn/fairlearn/pull/1353#issuecomment-2478323932)

Can you think of any other way to improve the example?

#### Suggest a potential alternative/fix
Replace `sklearn.ensemble.RandomForestClassifier` with `sklearn.ensemble.HistGradientBoostingClassifier` in the example for https://fairlearn.org/v0.12/user_guide/mitigation/reductions.html#exponentiated-gradient. 

File where the change needs to be made: https://github.com/fairlearn/fairlearn/blob/main/docs/user_guide/mitigation/reductions.rst
To build the documentation: https://fairlearn.org/main/contributor_guide/contributing_documentation.html

**Comment by milenaarneschedle** [comment]
Camila and I would like to work on this :)
